### DISCUSSION TITLE
Can I leave the flatribbon cable out?

### DISCUSSION THREAD

**Discussion by PaulAntonDeen** [gcormier/megadesk#100] [Q&A] [answered]
I have just received my megadesk (thanks for the quick shipment) and will be connecting it to ESPHOME with serial communication. There is no need for me to have the buttons so I will leave the original remote as is. and just glue the megadesk under my desk somewhere. Can megadesk run without the "key" that enables movement or do I need to solder a by-pass somewhere?

**Comment by gcormier** [comment, marked as answer]
The key activates a button on the actual ribbon cable itself that enables the ground signal.

Megadesk nor the OEM controller know about the key. The signal will merely be absent without it. (So it is also not possible to bypass the key using the factory ribbon cable/buttons.

![image](https://user-images.githubusercontent.com/8388983/162969965-ce0ddad5-fafb-4196-af1c-2dc2a464e79a.png)

**Reply by PaulAntonDeen** [reply]
Ah great so that means that if I am running it without manual input I do not need to worry about the "key"! Thanks

**Comment by PaulAntonDeen** [comment]
Can I program positions without the buttons. I have serial and ESPHome working so I assume I can move the desk to a height and then write a serial command to program the positions?

**Reply by gcormier** [reply]
Yes - commands are here https://github.com/gcormier/megadesk/blob/master/Serial.md

**Reply by PaulAntonDeen** [reply]
Nevermind, handled in home assistant by creating scripts that just set the desk height to a certain value, easier to change :)
Awesome product, thanks for the help!

**Reply by PaulAntonDeen** [reply]
In my case, because I only need two positions I ended up writing a template switch that just toggles between the low and high position and added this as a button entity in Home Assistant:

```
- platform: template
  switches:
    bureau:
      friendly_name: "Bureau"
      value_template: "{{states('number.megadesk_height_cm')|int > 100}}"
      icon_template: mdi:desk
      turn_on:
        - service: number.set_value
          data:
            value: '118'
          target:
            entity_id: number.megadesk_height_cm
      turn_off:
        - service: number.set_value
          data:
            value: '77'
          target:
            entity_id: number.megadesk_height_cm```
